src/game/Player.cpp

```diff
@@ -2778,10 +2778,10 @@ void Player::SendInitialSpells()
 
     uint16 spellCount = 0;
 
-    WorldPacket data(SMSG_INITIAL_SPELLS, (1 + 2 + 4 * m_spells.size() + 2 + m_spellCooldowns.size() * (2 + 2 + 2 + 4 + 4)));
+    WorldPacket data(SMSG_INITIAL_SPELLS, (1 + 2 + 6 * m_spells.size() + 2 + m_spellCooldowns.size() * (2 + 2 + 2 + 4 + 4)));
```

Comment:
Because the arithmetic there was wrong.  It was likely up-ported from a previous version of CMaNGOS, maybe the TBC version.  The structure of this packet has changed.  Note that for each spell, a uint32 and a uint16 is written.  Therefore, the buffer will need six bytes per spell, not four.  This is not a major issue, but it was causing a re-allocation that was un-necessary.